### PR TITLE
Upgrade okhttp to 5.1.0

### PR DESCRIPTION
**Pull request by ArnyminerZ**
There were some things changed on `MockWebServer`. Mainly that `MockResponse` now is immutable, so `MockResponse.Builder` exists; some function names have been changed.

Also `Response.body` now is never null, so in order to check whether a body has been set or not, I've checked whether the response length is `0`, and whether it has a content type. If none of this conditions apply, body is not set. This is done like this instead of simply checking for content length because when content type is not set we automatically assume it's XML.

### CODE REVIEW

File: src/main/kotlin/at/bitfire/dav4jvm/DavResource.kt [hunk 1]
```diff
@@ -729,34 +718,40 @@ open class DavResource @JvmOverloads constructor(
      *
      * @throws DavException if the response is not a Multi-Status response
      */
-    fun assertMultiStatus(response: Response) {
+    inline fun assertMultiStatus(response: Response, handleBody: (ResponseBody) -> Unit = {}) {
```
[the remark this answers is not in the document]
Comment (rfc2822) [reply src/main/kotlin/at/bitfire/dav4jvm/DavResource.kt:721]: Note: That's true, this will apply to -1 too.

**Review by sunkup — APPROVED** [review]
Yup. Looks good

File: src/main/kotlin/at/bitfire/dav4jvm/DavResource.kt [hunk 2]
```diff
@@ -722,9 +722,29 @@ open class DavResource @JvmOverloads constructor(
...
+    /**
+     * Checks whether the body of a [response] is set or not. This is done by checking the `Content-Length` header.
+     * - If it's `0`, and no `Content-Type` is set, no body is given.
+     * - If it's `-1` (not set), and `Transfer-Encoding` is not `chunked`, no body is given.
+     * Note that this last condition is necessary because chunked body requests doesn't have a `Content-Length`.
+     *
+     * This request will call [Response.body], but won't close it. It's responsibility of the caller to properly close it.
+     * @return `true` if [response] has a valid body, `false` otherwise.
+     */
+    private fun isEmptyResponseBody(response: Response): Boolean {
```
Comment (rfc2822) [inline src/main/kotlin/at/bitfire/dav4jvm/DavResource.kt:734]: The name indicates the opposite as what the method does:

- name: is empty response body
- return value: is valid response body

However, I think we can completely drop this method: as [response.body is now guaranteed to be non-null](https://github.com/square/okhttp/blob/d5b3a7471baf290e9040e6d579fa0844d23427bf/CHANGELOG.md?plain=1#L479-L482), we don't need the `?: throw Exception("empty response body")` anymore.

We can just process the body, and if we get an empty XML element, parsing will fail. We may however add a test to see how `processMultiStatus()` reacts to an empty body, and make sure that we get a useful exception in that case.

**Review by rfc2822 — COMMENTED** [review]
Very nice! Some comments:

1. In `assertMultiStatus`, we have~~~kotlin
val firstBytes = ByteArray(XML_SIGNATURE.size)
body.source().peek().readFully(firstBytes)
~~~

There's now `response.peekBody(size)` and we can directly use its return value to check the XML signature:

~~~kotlin
response.peekBody(XML_SIGNATURE.size).use {
  // …
}
~~~

2. We have some `httpResponse.body?.` in the code. Body is now non-nullable, so we can remove the `?`.

**Review by rfc2822 — APPROVED** [review]
